### PR TITLE
add OpenBSD to Eluant config dllmap

### PR DESCRIPTION
**Pull request by rfht**
simple addition of dllmap based on platform-specific Lua filenames.

### CODE REVIEW

File: thirdparty/Eluant.dll.config.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <configuration>
   <dllmap os="linux" dll="lua51.dll" target="@LIBLUA51@" />
+  <dllmap os="openbsd" dll="lua51.dll" target="liblua5.1.so" />
```
Comment (pchote) [inline thirdparty/Eluant.dll.config.in]: This probably wants to be using `@LIBLUA51@` for compatibility with the lua-detection script.

Comment (rfht) [reply thirdparty/Eluant.dll.config.in]: in that case I'll have to add some logic for filling in the correct `@LIBLUA51@` in those scripts. Slightly more complex, will work on it later.

Comment (pchote) [reply thirdparty/Eluant.dll.config.in]: We already have logic to do this (`thirdparty/configure-native-deps.sh`), and it must be working already - the config file isn't parsed at all if that fails.

**Comment by pchote** [comment]
#17038 has implemented the requested approach plus fixes for more recent regressions, so closing this one as superseded.

**Pull request closed without merging**